pbtranscript/ice/IceUtils.py

```
@@ -42,19 +42,24 @@
 
 def check_blasr(required_min_version=5.1):
     """
-    blasr version >= 5.1
+    blasr version >= required_min_version
     """
-    _o, _c, _e = backticks('blasr -version')
+    _o, _c, _e = backticks('blasr --version')
+    succeed = True
     if _c == 0:
         try:
             v = float('.'.join(_o[0].split()[1].split('.')[0:2]))
             assert(v >= required_min_version)
```

`assert` is the wrong way to do this. Assertions are suppressed when python is run with `-O`. Well, I guess that's ok here. We don't even build smrttools with `.pyo` files!

But it's better to add a message to your assertion. Then, you can simply let it propagate.

    def comparable(ver):
        for x in ver.strip().split('.')[:3]:
            try:
                yield int(x)
            except Exception:
                yield x
    cv = list(comparable(v))
    cr = list(comparable(req))
    assert canonical(v) >= , '{} < {}'.format(v, req)

I think that works. `[:3]` is to skip the 4th field in case that is a git SHA1. You could add a unit-test. It works because lists of unequal lengths can be compared.
```python
>>> [1,2] < [1,2]
False
>>> [1,2] < [1,3]
True
>>> [1,2] < [1,1]
False
>>> [1,2] < [1,1,0]
False
>>> [1,2] < [1,2,0]
True
>>> [1,2] < [1]
False
```